lib/request.ex

```diff
@@ -14,17 +14,29 @@ defmodule HTTPServer.Request do
   def parse_request(message) do
     [request_data | body] = message |> String.split("#{@carriage_return}#{@carriage_return}")
     [first | headers] = request_data |> String.split("#{@carriage_return}")
+    headers = parse_headers(headers, %{})
     [method, path, resource] = first |> String.split(" ")
 
     %__MODULE__{
       method: method,
       path: path,
       resource: resource,
-      headers: parse_headers(headers, %{}),
-      body: hd(body)
+      headers: headers,
+      body: parse_body(hd(body), headers)
```

Comment:
this is a little busy, would it make sense to move this up with the `parse_headers/2` call?

Reply:
Moved this up, but also changed some of the handling in parse_body/2 to accommodate Content-Types with parameters! Let me know if it still looks ok!